cli/azd/pkg/oneauth/oneauth.go

```diff
@@ -11,21 +11,25 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 )
 
-// Supported indicates whether this build includes OneAuth integration.
-const Supported = false
-
-var errNotSupported = errors.New("this build doesn't support OneAuth authentication")
-
-func LogIn(authority, clientID, scope string) (string, error) {
-	return "", errNotSupported
-}
-
-func Logout(clientID string) error {
-	return errNotSupported
-}
-
-func NewCredential(authority, clientID string, opts CredentialOptions) (azcore.TokenCredential, error) {
-	return nil, errNotSupported
-}
-
-func Shutdown() {}
+var (
+	// functions are defined as variables to make them easily replaceable in tests
```

Comment:
To be honest - I'm not in love with this pattern - I think the way we would have tested this would have been to create some interface that captured the behavior we wanted to override and then had a mock implementation in the test of this interface.  We'd then use our DI system to inject the production implementation in the AuthManager.

I feel like that would be overall nicer.

Reply:
Maybe so, but would that be a manual test i.e., would someone have to take the time to provision a DevBox and run the test suite on it?

Reply:
@chlowell We do have a series of manual end to end tests that we have our vendors run once a release - We could add a scenario here (I assume it could be as simple as provision a dev box, log in using one auth and then try to `azd up` an app?)

Reply:
Great, if you have vendors doing manual tests already, this would be a simple one to add. It would be as you describe, minus the logging in part because azd should infer the account from the OS. I'll back out this unit test.